modules/minimum-required-permissions-ipi-azure.adoc

```diff
@@ -0,0 +1,229 @@
+* `Microsoft.Compute/images/read`
+* `Microsoft.Compute/images/write`
+* `Microsoft.Compute/images/delete`
+====
+
+.Optional permissions for enabling user-defined disk encryption set
+[%collapsible]
+====
+* `Microsoft.Compute/diskEncryptionSets/read`
+* `Microsoft.Compute/diskEncryptionSets/write`
+* `Microsoft.Compute/diskEncryptionSets/delete`
+* `Microsoft.KeyVault/vaults/read`
+* `Microsoft.KeyVault/vaults/write`
+* `Microsoft.KeyVault/vaults/delete`
+* `Microsoft.KeyVault/vaults/deploy/action`
+* `Microsoft.KeyVault/vaults/keys/read`
+* `Microsoft.KeyVault/vaults/keys/write`
+* `Microsoft.Features/providers/features/register/action`
+====
+
+Following are the permissions required for deleting the {product-title} cluster in Azure public cloud.
+
+.Required permissions for deleting authorization resources
+[%collapsible]
+====
+* `Microsoft.Authorization/roleAssignments/delete`
+====
+
+.Required permissions for deleting compute resources
+[%collapsible]
+====
+* `Microsoft.Compute/disks/delete`
+* `Microsoft.Compute/galleries/delete`
+* `Microsoft.Compute/galleries/images/delete`
+* `Microsoft.Compute/galleries/images/versions/delete`
+* `Microsoft.Compute/virtualMachines/delete`
+====
+
+.Required permissions for deleting identity management resources
+[%collapsible]
+====
+* `Microsoft.ManagedIdentity/userAssignedIdentities/delete`
+====
+
+.Required permissions for deleting network resources
+[%collapsible]
+====
+* `Microsoft.Network/dnszones/read`
+* `Microsoft.Network/dnsZones/A/read`
+* `Microsoft.Network/dnsZones/A/delete`
+* `Microsoft.Network/dnsZones/CNAME/read`
+* `Microsoft.Network/dnsZones/CNAME/delete`
+* `Microsoft.Network/loadBalancers/delete`
+* `Microsoft.Network/networkInterfaces/delete`
+* `Microsoft.Network/networkSecurityGroups/delete`
+* `Microsoft.Network/privateDnsZones/read`
+* `Microsoft.Network/privateDnsZones/A/read`
+* `Microsoft.Network/privateDnsZones/delete`
+* `Microsoft.Network/privateDnsZones/virtualNetworkLinks/delete`
+* `Microsoft.Network/publicIPAddresses/delete`
+* `Microsoft.Network/virtualNetworks/delete`
+====
+
+.Required permissions for checking health of resources
+[%collapsible]
+====
+* `Microsoft.Resourcehealth/healthevent/Activated/action`
+* `Microsoft.Resourcehealth/healthevent/Resolved/action`
+* `Microsoft.Resourcehealth/healthevent/Updated/action`
+====
+
+.Required permissions for deleting resource groups
+[%collapsible]
+====
+* `Microsoft.Resources/subscriptions/resourcegroups/delete`
+====
+
+.Required permissions for deleting storage resources
+[%collapsible]
+====
+* `Microsoft.Storage/storageAccounts/delete`
+* `Microsoft.Storage/storageAccounts/listKeys/action`
+====
+
+[NOTE]
+====
+You can create a custom role with all the permissions and assign that custom role to the service principal. To create a custom role in the Azure portal, see link:https://learn.microsoft.com/en-us/azure/role-based-access-control/custom-roles[Azure custom roles] in the Azure documentation.
+====
```

Comment:
@jinyunma as discussed over slack, we will add a separate Note here. 

```
- For cluster installation, the permissions should be scoped to subscription level, and later on can be re-scoped to the installer created resource group level. If the Public DNS zone is present in different resource group, then Network DNS zone related permissions should always be subscription level.

- Azure identity will get Contributor role by the installer.

- For cluster deletion, all the permissions can be scoped to subscription level.

```

This is be added only for IPI @xenolinux FYI

Reply:
@chiragkyal We are removing the existing note and adding a new note right?

Reply:
We will add a new note just after the existing note.

Reply:
Addressed